ossdbtoolsservice/driver/types/pymysql_driver.py

```diff
@@ -346,3 +343,23 @@ def handle_connection_error(self, exception: pymysql.err.Error):
                 if "Access denied for user" in message:
                     raise OssdbToolsServiceException(OssdbErrorCodes.MYSQL_FLEX_INCORRECT_CREDENTIALS(code, message))
         raise exception
+    
+    def _set_ssl_options(self, conn_params: dict):
+        ssl_mode = MySQLSSLMode[self._connection_options["ssl"]] if "ssl" in self._connection_options else DEFAULT_SSL_MODE
+        if ssl_mode == MySQLSSLMode.disable:
+            self._connection_options["ssl"] = None
+        else:
+            # Fill up the ssl dict
+            ssl_params = {param for param in conn_params if param.startswith("ssl.")}
+            ssl_dict = {param.strip("ssl."): conn_params[param] for param in ssl_params}
+            
+            if ssl_mode.value <= MySQLSSLMode.require.value:
+                # ca is not required for modes less than equal to require
+                ssl_dict['ca'] = None
+            elif 'ca' not in ssl_dict or ssl_dict['ca'] is None :
+                # Raise error is ca is not provided for verify modes
+                raise OssdbToolsServiceException(OssdbErrorCodes.MYSQL_SSL_CA_REQUIRED_FOR_VERIFY_MODE())
+                  
+            ssl_dict["verify_mode"] = "required" if ssl_mode.value >= MySQLSSLMode.verify_ca.value else "none"
+            ssl_dict["check_hostname"] = True if ssl_mode == MySQLSSLMode.verify_identity else False
```

Comment:
Minor: Need to understand the explicit failure scenario for  "check_hostname" mode.

Reply:
During ssl handshake, server sends its identity along with the certificate. This identity is checked by the client if check hostname is set to true.
This helps prevents man in middle attack.